assets/runtime/functions

```diff
@@ -1048,8 +1048,27 @@ migrate_database() {
   CACHE_VERSION=
   [[ -f ${REDMINE_DATA_DIR}/tmp/VERSION ]] && CACHE_VERSION=$(cat ${REDMINE_DATA_DIR}/tmp/VERSION)
   if [[ ${REDMINE_VERSION} != ${CACHE_VERSION} ]]; then
+    if [[ "${DB_CREATE:-true}" == "true" ]] ; then
+      echo "Creating database..."
+      # Disable "exit on error" so we can capture the error code
+      set +e
+      exec_as_redmine bundle exec rake db:create >/dev/null
+      res=$?
+      # Restore "exit on error"
+      set -e
+
+      if [[ 0 -ne $res ]] ; then
```

Comment:
Note shellcheck will tell you its better to if check the command directly instead of storing the results.  I think this also would mean we don't have to disable exit on error.

```suggestion
      if ! exec_as_redmine bundle exec rake db:create >/dev/null
      then
```

Reply:
> Note shellcheck will tell you its better to if check the command directly instead of storing the results. I think this also would mean we don't have to disable exit on error.

I wanted to retain the exit value from `rake db:create`, since I don't know the range of non-zero values that it can produce. The only way I know of doing that is to do the `res=$?`, which requires disabling "exit-on-error".

I'm happy to change it you are ok with returning a fixed value, e.g.:
```
if ! exec_as_redmine bundle exec rake db:create >/dev/null ; then
  echo <<-EOM
Failed to create db. If db exists, and user (or role) exists and has access to db but no permission to create databases, set:
-----------------------
DB_CREATE=false
-----------------------
EOM

       # FWIW, using "return" here would be consistent a similar use of "exec_as_redmine' in "redmine_check_database_connection"
        exit 1
fi
```

Reply:
> I'm happy to change it you are ok with returning a fixed value, e.g.:

I think a fixed error code is worth the simplified code.

>        # FWIW, using "return" here would be consistent a similar use of "exec_as_redmine' in "redmine_check_database_connection"
Good point. Entrypoint.sh does use set -e so returning an error should still bail the setup like it did before this change.